test/formal/checked_int_mul_16.py

```diff
@@ -26,16 +26,23 @@
 	# cast to full n_bits values
 	X = BVSignedUpCast(X_short, n_bits)
 	Y = BVSignedUpCast(Y_short, n_bits)
+	product = MUL(X, Y)
 
 	# Constants
 	maxValue = BVSignedMax(type_bits, n_bits)
 	minValue = BVSignedMin(type_bits, n_bits)
 
 	# Overflow and underflow checks in YulUtilFunction::overflowCheckedIntMulFunction
-	overflow_check_1 = AND(AND(SGT(X, 0), SGT(Y, 0)), GT(X, DIV(maxValue, Y)))
-	underflow_check_1 = AND(AND(SGT(X, 0), SLT(Y, 0)), SLT(Y, SDIV(minValue, X)))
-	underflow_check_2 = AND(AND(SLT(X, 0), SGT(Y, 0)), SLT(X, SDIV(minValue, Y)))
-	overflow_check_2 = AND(AND(SLT(X, 0), SLT(Y, 0)), SLT(X, SDIV(maxValue, Y)))
+	if type_bits == 256:
```

Comment:
This should probably be:
```suggestion
	if type_bits == n_bits:
```

Here and in the other ones as well.

The underlying problem here is that the SMT solvers don't live up to proving a lot of these up to bit vectors as large as 256 bits - so we usually have to reduce the bit size and argue that it doesn't matter, if you stop at 256 bits or at 16 bits :-).
For the multiplication overflow checks - depending on what we end up doing in the end - we should take care that we have a case where ``n_bits/2 < type_bits < n_bits``, though.
We may need to change ``type_bits *= 2`` to ``type_bits += 4`` below or something like that.

Reply:
Done.